src/core/thread.d

```diff
@@ -2574,12 +2511,10 @@ extern (C) void thread_suspendAll() nothrow
             {
                 Thread.remove(t);
             }
-            else if (t.m_isInCriticalRegion)
+            else if (atomicLoad(t.m_isInCriticalRegion))
```

Comment:
This could probably be done with acquire semantics, which will provide a good speedup on Intel.